### PR TITLE
Test validation of increment and decrement statements.

### PR DESCRIPTION
**Pull request by dneto0**
Bug: #1646

Issue: #1646

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [x] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/shader/validation/statement/increment_decrement.spec.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  v3u_y: { type: 'vec3u', wgsl: 'a.y', pass: true },
+  v3u_z: { type: 'vec3u', wgsl: 'a.z', pass: true },
+  v4u_x: { type: 'vec4u', wgsl: 'a.x', pass: true },
+  v4u_y: { type: 'vec4u', wgsl: 'a.y', pass: true },
+  v4u_z: { type: 'vec4u', wgsl: 'a.z', pass: true },
+  v4u_w: { type: 'vec4u', wgsl: 'a.w', pass: true },
+  v2i_x: { type: 'vec2i', wgsl: 'a.x', pass: true },
+  v2i_y: { type: 'vec2i', wgsl: 'a.y', pass: true },
+  v3i_x: { type: 'vec3i', wgsl: 'a.x', pass: true },
+  v3i_y: { type: 'vec3u', wgsl: 'a.y', pass: true },
```
Comment (dj2) [inline src/webgpu/shader/validation/statement/increment_decrement.spec.ts:94]:
```suggestion
  v3i_y: { type: 'vec3i', wgsl: 'a.y', pass: true },
```

File: src/webgpu/shader/validation/statement/increment_decrement.spec.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  v3i_z: { type: 'vec3i', wgsl: 'a.z', pass: true },
+  v4i_x: { type: 'vec4i', wgsl: 'a.x', pass: true },
+  v4i_y: { type: 'vec4u', wgsl: 'a.y', pass: true },
```
Comment (dj2) [inline src/webgpu/shader/validation/statement/increment_decrement.spec.ts:97]:
```suggestion
  v4i_y: { type: 'vec4i', wgsl: 'a.y', pass: true },
```

File: src/webgpu/shader/validation/parse/increment_decrement.spec.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  },
+  sampler: { wgsl: 'g++;', pass: false, gdecl: '@group(0) @binding(0) var g: sampler;' },
+  sampler_comparison: {
+    wgsl: 'g++;',
+    pass: false,
+    gdecl: '@group(0) @binding(0) var g: sampler_comparison;',
+  },
+  override: { wgsl: 'g++;', pass: false, gdecl: 'override g:i32;' },
+  global_const: { wgsl: 'g++;', pass: false, gdecl: 'const g:i32 = 0;' },
+
```
[the remark this answers is not in the document]
Comment (dneto0) [reply src/webgpu/shader/validation/parse/increment_decrement.spec.ts:96]: done

File: src/webgpu/shader/validation/parse/increment_decrement.spec.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+}
+
+const kTests = {
+  var: { wgsl: 'a++;', pass: true },
+  vector: { wgsl: 'v++;', pass: false },
+  paren_var_paren: { wgsl: '(a)++;', pass: true },
+  star_and_var: { wgsl: '*&a++;', pass: true },
+  paren_star_and_var_paren: { wgsl: '(*&a)++;', pass: true },
+  many_star_and_var: { wgsl: '*&*&*&a++;', pass: true },
+
```
[the remark this answers is not in the document]
Comment (dneto0) [reply src/webgpu/shader/validation/parse/increment_decrement.spec.ts:25]: done.  inc_space_inc  inc_paren_inc

File: src/webgpu/shader/validation/statement/increment_decrement.spec.ts (hunk 1 above)
Comment (dneto0) [reply src/webgpu/shader/validation/statement/increment_decrement.spec.ts:94]: done

File: src/webgpu/shader/validation/statement/increment_decrement.spec.ts (hunk 2 above)
Comment (dneto0) [reply src/webgpu/shader/validation/statement/increment_decrement.spec.ts:97]: done

File: src/webgpu/shader/validation/statement/increment_decrement.spec.ts (hunk 1 above)
Comment (dj2) [reply src/webgpu/shader/validation/statement/increment_decrement.spec.ts:94]: Forgot to commit?

**Review by dj2 — APPROVED** [review]

**Pull request merged**